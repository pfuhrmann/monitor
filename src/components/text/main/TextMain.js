import React from 'react';
import PropTypes from 'prop-types';
import './TextMain.scss';

const TextMain = ({ children }) => (
  <h1>{children}</h1>
);

TextMain.propTypes = {
  children: PropTypes.node,
};

TextMain.defaultProps = {
  children: '',
};

export default TextMain;
