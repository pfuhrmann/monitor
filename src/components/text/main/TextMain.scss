@import "../../../variables.scss";

h2 {
  color: $color-catalina-blue;
  font-size: 1.3125rem;
  font-weight: $font-weight-semi-bold;
}
