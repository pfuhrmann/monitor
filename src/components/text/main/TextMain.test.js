import React from 'react';
import ReactDOM from 'react-dom';
import TextMain from './TextMain';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<TextMain />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly h1 tag with text', () => {
  const component = renderer.create(
    <TextMain>Header</TextMain>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
