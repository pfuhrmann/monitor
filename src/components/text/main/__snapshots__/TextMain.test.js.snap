// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly h1 tag with text 1`] = `
<h1>
  Header
</h1>
`;
